src/tools/mediaCapabilitiesProber/api/probeMediaConfiguration.ts

```diff
@@ -0,0 +1,123 @@
+/**
+ * Copyright 2015 CANAL+ Group
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import getProbedConfiguration, { ICapabilitiesTypes } from "../capabilities";
+import probers from "../probers";
+import { IMediaConfiguration } from "../types";
+
+import log from "../../../utils/log";
+import filterEmptyFields from "../utils/filterEmptyFields";
+import isEmpty from "../utils/isEmpty";
+
+type IBrowserAPIS =
+  "_isTypeSupported_" |
+  "_isTypeSupportedWithFeatures_" |
+  "_matchMedia_" |
+  "_decodingInfos_" |
+  "_requestMediaKeySystemAccess_" |
+  "_getStatusForPolicy_";
+
+/**
+ * Assert that configuration is valid before probing:
+ * 1 - Filter empty fields.
+ * 2 - Check for emptyness.
+ * 3 - Check for unsupported configuration attributes.
+ * @param {Object} config
+ */
+const validateConfiguration = (config: IMediaConfiguration) => {
+  if (!config) {
+    throw new Error("MCP_CONF: Configuration is not defined.");
+  }
+  const filteredConfig: IMediaConfiguration = filterEmptyFields(config);
+  if (isEmpty(filteredConfig)) {
+    throw new Error("MCP_CONF: Can't probe empty configuration.");
+  }
+  return filteredConfig;
+};
+
+/**
+ * Probe media capabilities, evaluating capabilities with available browsers API.
+ *
+ * Probe every given features with configuration.
+ * If the browser API is not available OR we can't call browser API with enough arguments,
+ * do nothing but warn user (e.g. HDCP is not specified for calling "getStatusForPolicy"
+ * API, "mediaCapabilites" API is not available.).
+ *
+ * if we call the browser API, we get from it a number which means:
+ * - 0 : Probably
+ * - 1 : Maybe
+ * - 2 : Not Supported
+ *
+ * From all API results, we return worst of states (e.g. if one API returns
+ * "Not Supported" among "Probably" statuses, return "Not Supported").
+ *
+ * If no API was called or some capabilites could not be probed and status is "Probably",
+ * return "Maybe".
+ * @param {Object} config
+ */
+
+const browserAPIS: IBrowserAPIS[] = [
+  "_isTypeSupported_",
+  "_isTypeSupportedWithFeatures_",
+  "_matchMedia_",
+  "_decodingInfos_",
+  "_requestMediaKeySystemAccess_",
+  "_getStatusForPolicy_",
+];
+
+const probeMediaConfiguration =
```

Comment:
JSDoc :p

Reply:
Also why an arrow function instead of a regular one? The rest of the code is not like that.